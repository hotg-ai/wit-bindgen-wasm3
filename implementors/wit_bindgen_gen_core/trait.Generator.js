(function() {var implementors = {};
implementors["wit_bindgen_gen_wasm3"] = [{"text":"impl <a class=\"trait\" href=\"wit_bindgen_gen_core/trait.Generator.html\" title=\"trait wit_bindgen_gen_core::Generator\">Generator</a> for <a class=\"struct\" href=\"wit_bindgen_gen_wasm3/struct.Wasmtime.html\" title=\"struct wit_bindgen_gen_wasm3::Wasmtime\">Wasmtime</a>","synthetic":false,"types":["wit_bindgen_gen_wasm3::Wasmtime"]}];
if (window.register_implementors) {window.register_implementors(implementors);} else {window.pending_implementors = implementors;}})()