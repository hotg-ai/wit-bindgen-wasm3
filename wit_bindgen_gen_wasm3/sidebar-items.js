initSidebarItems({"enum":[["Async",""]],"struct":[["Opts",""],["Wasmtime",""]]});