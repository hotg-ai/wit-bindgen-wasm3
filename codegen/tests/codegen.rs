#[macro_use]
#[cfg(test)]
extern crate pretty_assertions;

use std::{collections::HashMap, path::Path};

use wit_bindgen_gen_core::{wit_parser::Interface, Files, Generator};
use wit_bindgen_gen_wasm3::Opts;

#[test]
fn single_function() {
    let path = Path::new(env!("CARGO_MANIFEST_DIR"))
        .join("tests")
        .join("data")
        .join("single-function.wit");
    let single_function = Interface::parse_file(&path).unwrap();
    let mut files = Files::default();

    Opts::default()
        .build()
        .generate_all(&[single_function], &[], &mut files);

    let files: HashMap<_, _> = files.iter().collect();
    assert_eq!(files.len(), 1);
    let generated = std::str::from_utf8(files["bindings.rs"]).unwrap();
    println!("=========\n{}=======", generated);
    assert_eq!(generated, "asd");
}
