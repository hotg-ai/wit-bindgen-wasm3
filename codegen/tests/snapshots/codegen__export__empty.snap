---
source: codegen/tests/codegen.rs
expression: empty
---
pub mod empty {
    #[allow(unused_imports)]
    use wit_bindgen_wasm3::rt::{anyhow, wasm3};
}

